internal/cluster/cluster.go

```diff
@@ -562,3 +567,38 @@ func (m *Manager) IsReadyForInstallation(c *common.Cluster) (bool, string) {
 	}
 	return true, ""
 }
+
+func (m *Manager) SetConnectivityNajorityGroupsForCluster(c *common.Cluster, db *gorm.DB) error {
+	if db == nil {
+		db = m.db
+	}
+	if !funk.ContainsString([]string{models.ClusterStatusPendingForInput, models.ClusterStatusInsufficient, models.ClusterStatusReady}, swag.StringValue(c.Status)) {
+		return nil
+	}
+	var cluster common.Cluster
+	if err := db.Preload("Hosts", "status <> ? and connectivity <> ''", models.HostStatusDisabled).Take(&cluster, "id = ?", c.ID.String()).Error; err != nil {
+		return common.NewApiError(http.StatusBadRequest, errors.Wrapf(err, "Getting cluster %s", c.ID.String()))
+	}
+	majorityGroups := make(map[string][]strfmt.UUID)
```

Comment:
This looks like a fairly expensive calculation that we repeat often, with the same inputs - worth caching.

Reply:
OK

Reply:
Delayed for later performance improvement task. https://issues.redhat.com/browse/MGMT-2446